e2e/tests/node.go

```diff
@@ -178,8 +181,21 @@ func (p *pocketdBin) runCurlCmd(rpcBaseURL, service, method, path, appAddr, data
 		return nil, err
 	}
 
+	// Store the virtual host we want to send in the Host header
+	virtualHost := rpcUrl.Host
 	if len(service) > 0 {
-		rpcUrl.Host = fmt.Sprintf("%s.%s", service, rpcUrl.Host)
+		// Strip port if it exists and add service prefix
```

Comment:
Can you create a helper called "getVirtualHostFromHost" w/ a longer explanation of what's going on here?